toolchain/check/import_cpp.cpp

```diff
@@ -446,18 +461,27 @@ static auto ImportFunctionDecl(Context& context, SemIR::LocId loc_id,
     context.TODO(loc_id, "Unsupported: Template function");
     return SemIR::ErrorInst::InstId;
   }
+  context.pattern_block_stack().Push();
   auto param_patterns_id =
       MakeParamPatternsBlockId(context, loc_id, *clang_decl);
   if (!param_patterns_id.has_value()) {
+    context.pattern_block_stack().Pop();
     return SemIR::ErrorInst::InstId;
   }
   auto return_slot_pattern_id = GetReturnType(context, loc_id, clang_decl);
+  auto pattern_block_id = context.pattern_block_stack().Pop();
   if (SemIR::ErrorInst::InstId == return_slot_pattern_id) {
     return SemIR::ErrorInst::InstId;
   }
 
+  context.inst_block_stack().Push();
```

Comment:
I think you need to move this up, at least above `GetReturnType`, but preferably to the beginning of the function, because this block needs to hold all the non-pattern insts emitted as part of the function declaration. Right now the type insts for the return type are getting lost.

(Those insts don't really belong here, but we don't have a better place for them yet, so this code should be consistent with the native-Carbon code).